src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectReportController.java

```diff
@@ -0,0 +1,128 @@
+
+    @Value("${pa.catalog.security.required.sessionid}")
+    private boolean sessionIdRequired;
+
+    @ApiOperation(value = "Get list of catalog objects in a PDF report file")
+    @ApiResponses(value = { @ApiResponse(code = 401, message = "User not authenticated"),
+                            @ApiResponse(code = 403, message = "Permission denied"), })
+    @RequestMapping(method = GET)
+    @ResponseStatus(HttpStatus.OK)
+    public void getReport(HttpServletResponse response,
+            @ApiParam(value = "sessionID", required = false) @RequestHeader(value = "sessionID", required = false) String sessionId,
+            @ApiParam(value = "The name of the user who owns the Bucket") @RequestParam(value = "owner", required = false) String ownerName,
+            @ApiParam(value = "The kind of objects that buckets must contain") @RequestParam(value = "kind", required = false) Optional<String> kind,
+            @ApiParam(value = "The content type of objects that buckets must contain") @RequestParam(value = "contentType", required = false) Optional<String> contentType)
+            throws NotAuthenticatedException, AccessDeniedException, IOException {
+
+        List<String> authorisedBucketsNames = getListOfAuthorizedBuckets(sessionId, ownerName, kind, contentType);
+
+        byte[] content = catalogObjectReportService.generateBytesReport(authorisedBucketsNames, kind, contentType);
+
+        response.addHeader("Content-size", new Integer(content.length).toString());
+        response.setCharacterEncoding("UTF-8");
+
+        response.getOutputStream().write(content);
+        response.getOutputStream().flush();
+
+    }
+
+    private List<String> getListOfAuthorizedBuckets(String sessionId, String ownerName, Optional<String> kind,
+            Optional<String> contentType) throws NotAuthenticatedException, AccessDeniedException {
+        List<BucketMetadata> authorisedBuckets;
+        if (sessionIdRequired) {
+            RestApiAccessResponse restApiAccessResponse = restApiAccessService.checkAccessBySessionIdForOwnerOrGroupAndThrowIfDeclined(sessionId,
+                                                                                                                                       ownerName);
+
+            authorisedBuckets = bucketService.getBucketsByGroups(sessionId,
+                                                                 ownerName,
+                                                                 kind,
+                                                                 contentType,
+                                                                 () -> restApiAccessResponse.getAuthenticatedUser()
+                                                                                            .getGroups());
+
+        } else {
+            authorisedBuckets = bucketService.listBuckets(ownerName, kind, contentType);
+        }
+
+        List<String> authorisedBucketsNames = authorisedBuckets.stream()
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "authorisedBucketsNames". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1488)